src/kokkos_ext/ArborX_DetailsKokkosExtViewHelpers.hpp

```diff
@@ -104,21 +89,14 @@ typename View::non_const_type clone(ExecutionSpace const &space, View const &v)
 
 template <class ExecutionSpace, class View>
 typename View::non_const_type
-cloneWithoutInitializingNorCopying(ExecutionSpace const &space, View const &v,
-                                   std::string const &label)
+cloneWithoutInitializingNorCopying(ExecutionSpace const &space, View const &v)
 {
   static_assert(Kokkos::is_execution_space<ExecutionSpace>::value);
   static_assert(Kokkos::is_view<View>::value);
-  return typename View::non_const_type(
-      Kokkos::view_alloc(space, Kokkos::WithoutInitializing, label),
-      v.layout());
-}
-
-template <class ExecutionSpace, class View>
-typename View::non_const_type
-cloneWithoutInitializingNorCopying(ExecutionSpace const &space, View const &v)
-{
-  return cloneWithoutInitializingNorCopying(space, v, v.label());
+  return Kokkos::create_mirror(Kokkos::view_alloc(typename View::memory_space{},
+                                                  space,
```

Comment:
This seems to do something different from what it used to. In the previous iteration, the memory space was taken from the execution space, now it is taken from the view. Not saying it's wrong, just a change.

Reply:
Yes, that's fair. To me, the function reads "copy the View with all its properties" using the execution space given. If we take the memory space from the execution space, the memory space might change (especially for Views in shared memory spaces).